agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -140,6 +179,32 @@ def ensure_started
       raise exc
     end
 
+    def ensure_healing
+      if health.count < 5
+        info "restarting #{service_pod.name} because it's unhealthy"
+        restart_instance
+      elsif health.count == 5
+        info "re-creating #{service_pod.name} because it's unhealthy"
+        ensure_running(pull: false)
+      elsif health.count % 5 == 0
```

Comment:
Wut, this `health.count % 5` condition seems very odd... how this will behave depends on what the relative phase of the `ServicePodManager` loop and the `ContainerHealthCheckWorker` interval loop? i.e. I have no idea how this will behave? It might not actually do anything, because the `else` condition is a no-op?